[Bug] <GMANGA>: <EXTENSION CRASHES>

## Device information
- Tachiyomi version:0.10.7 
-Android version: 6.0 


## Source information
- Name: gmanga
- Extension version: 1.2.2

## Steps to reproduce
1. First Step: open gmanga source


### Expected behavior
the source open normally

### Actual behavior
the app CRASHES and close itself.

## Other details
If I open "latest" on gmanga source ,it show latest manga  normally but when opening any one from latest manga the app CRASHES!

<details>
  <summary>logcat</summary>

```

[01-06 17:18:00.648 12317:12477 E/ACRA]
ACRA caught a IllegalStateException for eu.kanade.tachiyomi
java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:59)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Ljava/util/Base64;
	at eu.kanade.tachiyomi.extension.ar.gmanga.GmangaCryptoUtilsKt.aesDecrypt(GmangaCryptoUtils.kt:41)
	at eu.kanade.tachiyomi.extension.ar.gmanga.GmangaCryptoUtilsKt.decrypt(GmangaCryptoUtils.kt:15)
	at eu.kanade.tachiyomi.extension.ar.gmanga.Gmanga.decryptResponse(Gmanga.kt:164)
	at eu.kanade.tachiyomi.extension.ar.gmanga.Gmanga.chapterListParse(Gmanga.kt:67)
	at eu.kanade.tachiyomi.source.online.HttpSource$fetchChapterList$1.call(HttpSource.kt:214)
	at eu.kanade.tachiyomi.source.online.HttpSource$fetchChapterList$1.call(HttpSource.kt:26)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:69)
	at rx.internal.operators.OnSubscribeDoOnEach$DoOnEachSubscriber.onNext(OnSubscribeDoOnEach.java:101)
	at eu.kanade.tachiyomi.network.OkHttpExtensionsKt$asObservable$1$requestArbiter$1.request(OkHttpExtensions.kt:30)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber$1.request(OperatorSubscribeOn.java:109)
	at rx.Subscriber.setProducer(Subscriber.java:211)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.setProducer(OnSubscribeMap.java:102)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.setProducer(OperatorSubscribeOn.java:105)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.setProducer(OnSubscribeMap.java:102)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at eu.kanade.tachiyomi.network.OkHttpExtensionsKt$asObservable$1.call(OkHttpExtensions.kt:50)
	at eu.kanade.tachiyomi.network.OkHttpExtensionsKt$asObservable$1.call(OkHttpExtensions.kt:18)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:41)
	at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:51)
	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:35)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:230)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 7 more
Caused by: java.lang.ClassNotFoundException: Didn't find class "java.util.Base64" on path: DexPathList[[zip file "/data/app/eu.kanade.tachiyomi.extension.ar.gmanga-1/base.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
	... 39 more
	Suppressed: java.lang.ClassNotFoundException: Didn't find class "java.util.Base64" on path: DexPathList[[zip file "/data/app/eu.kanade.tachi

[01-06 17:18:00.975 195:631 I/BufferQueueProducer]
[eu.kanade.tachiyomi/eu.kanade.tachiyomi.ui.main.MainActivity](this:0xb5c55000,id:4691,api:1,p:12317,c:195) queueBuffer: fps=12.85 dur=1011.91 max=426.47 min=16.37

[01-06 17:18:00.997 12317:12317 W/Glide]
Load failed for MangaThumbnail(manga=eu.kanade.tachiyomi.data.database.models.LibraryManga@7fe8b876, coverLastModified=0) with size [334x480]
class com.bumptech.glide.load.engine.GlideException: Failed to load resource
There was 1 cause:
com.bumptech.glide.load.HttpException()
 call GlideException#logRootCauses(String) for more detail
  Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, LOCAL
There was 1 cause:
com.bumptech.glide.load.HttpException()
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 1): class com.bumptech.glide.load.HttpException: 

```

</details>

--- a/src/ar/gmanga/build.gradle
+++ b/src/ar/gmanga/build.gradle
@@ -5,7 +5,7 @@ ext {
     extName = 'GMANGA'
     pkgNameSuffix = 'ar.gmanga'
     extClass = '.Gmanga'
-    extVersionCode = 3
+    extVersionCode = 2
     libVersion = '1.2'
     containsNsfw = false
 }

--- a/src/ar/gmanga/build.gradle
+++ b/src/ar/gmanga/build.gradle
@@ -5,7 +5,7 @@ ext {
     extName = 'GMANGA'
     pkgNameSuffix = 'ar.gmanga'
     extClass = '.Gmanga'
-    extVersionCode = 2
+    extVersionCode = 4
     libVersion = '1.2'
     containsNsfw = false
 }

--- a/src/ar/gmanga/src/eu/kanade/tachiyomi/extension/ar/gmanga/Gmanga.kt
+++ b/src/ar/gmanga/src/eu/kanade/tachiyomi/extension/ar/gmanga/Gmanga.kt
@@ -110,7 +110,7 @@ class Gmanga : ConfigurableSource, HttpSource() {
     }
 
     override fun latestUpdatesRequest(page: Int): Request {
-        return GET("$baseUrl/releases", headers)
+        return GET("$baseUrl/mangas/latest", headers)
     }
 
     override fun mangaDetailsParse(response: Response): SManga {

--- a/src/ar/gmanga/src/eu/kanade/tachiyomi/extension/ar/gmanga/Gmanga.kt
+++ b/src/ar/gmanga/src/eu/kanade/tachiyomi/extension/ar/gmanga/Gmanga.kt
@@ -101,8 +101,11 @@ class Gmanga : ConfigurableSource, HttpSource() {
                 SManga.create().apply {
                     url = "/mangas/${it["id"].asString}"
                     title = it["title"].asString
-                    val thumbnail = "medium_${it["cover"].asString.substringBeforeLast(".")}.webp"
-                    thumbnail_url = "https://media.$domain/uploads/manga/cover/${it["id"].asString}/$thumbnail"
+
+                    thumbnail_url = it["cover"].nullString?.let { coverFileName ->
+                        val thumbnail = "medium_${coverFileName.substringBeforeLast(".")}.webp"
+                        "https://media.$domain/uploads/manga/cover/${it["id"].asString}/$thumbnail"
+                    }
                 }
             },
             false

--- a/src/ar/gmanga/src/eu/kanade/tachiyomi/extension/ar/gmanga/GmangaCryptoUtils.kt
+++ b/src/ar/gmanga/src/eu/kanade/tachiyomi/extension/ar/gmanga/GmangaCryptoUtils.kt
@@ -1,9 +1,7 @@
 package eu.kanade.tachiyomi.extension.ar.gmanga
 
-import android.annotation.TargetApi
-import android.os.Build
+import android.util.Base64
 import java.security.MessageDigest
-import java.util.Base64
 import javax.crypto.Cipher
 import javax.crypto.spec.IvParameterSpec
 import javax.crypto.spec.SecretKeySpec
@@ -36,14 +34,12 @@ private fun String.sha256(): String {
         .fold("", { str, it -> str + "%02x".format(it) })
 }
 
-@TargetApi(Build.VERSION_CODES.O)
 private fun String.aesDecrypt(secretKey: ByteArray, ivString: String): String {
-    val decoder = Base64.getDecoder()
     val c = Cipher.getInstance("AES/CBC/PKCS5Padding")
     val sk = SecretKeySpec(secretKey, "AES")
-    val iv = IvParameterSpec(decoder.decode(ivString.toByteArray(Charsets.UTF_8)))
+    val iv = IvParameterSpec(Base64.decode(ivString.toByteArray(Charsets.UTF_8), Base64.DEFAULT))
     c.init(Cipher.DECRYPT_MODE, sk, iv)
 
-    val byteStr = decoder.decode(this.toByteArray(Charsets.UTF_8))
+    val byteStr = Base64.decode(this.toByteArray(Charsets.UTF_8), Base64.DEFAULT)
     return String(c.doFinal(byteStr))
 }

